tests/common/factories/group.py

```diff
@@ -58,3 +47,14 @@ class RestrictedGroup(Group):
 
     joinable_by = None
     readable_by = ReadableBy.world
+
+
+class GroupMembership(ModelFactory):
+    class Meta:
+        model = models.GroupMembership
+        exclude = ("user", "group")
+
+    user = factory.SubFactory(User)
+    user_id = factory.LazyAttribute(lambda membership: membership.user.id)
+    group = factory.SubFactory(Group)
+    group_id = factory.LazyAttribute(lambda membership: membership.group.id)
```

Comment:
I've added a group factory in order to make it easier for tests to add users to groups. Tests can just do:

```python
factories.GroupMembership(user=user, group=group)
```

1. Tests don't have to have the `db_session` and add the `GroupMembership` to it
2. Tests can pass user and group objects directly to `factories.GroupMembership()`, whereas `models.GroupMembership()` would require you to pass the `user_id` and `group_id` instead which is slightly more awkward. (This should be fixed in future by adding `user` and `group` relationships to `models.GroupMembership` but that can't be done right now: it needs to be done at the same time as replacing `User.groups` and `Group.members` with `User.memberships` and `Group.memberships`, you can't have both the old `groups` and `members` relationships and the new `memberships` ones at the same time, doing so seems to trigger SQLAlchemy warnings.)

For now the factory needs the `exclude = ("user", "group")` because `user` and `group` aren't actually attributes of `models.GroupMembership`, they're only used by the factory to generate `user_id` and `group_id`.

Reply:
FIXME: I think if you do `factories.GroupMembership(user_id=..., group_id=...)` this will still create `User` and `Group` objects but not use them. Might want to fix that